recipes/mosquitto/2.x/conanfile.py

```diff
@@ -0,0 +1,136 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+class Mosquitto(ConanFile):
+    name = "mosquitto"
+    license = "EPL-2.0", "EPL-1.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://mosquitto.org"
+    description = """Eclipse Mosquitto MQTT library, broker and more"""
+    topics = ("MQTT", "IoT", "eclipse")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+                "with_tls": [True, False],
+                "clients": [True, False],
+                "broker": [True, False],
+                "apps": [True, False],
+                "plugins": [True, False],
+#                 "with_cjson": [True, False], TODO , adding a dependency is not enough, it needs also be found
```

Comment:
```suggestion
                 "with_cjson": [True, False], TODO , adding a dependency is not enough, it needs also be found
```

You can enable it and raise an exception when True, so you don't need to add a new option in the future and change the package ID.

Reply:
but it should have a different package id, since it will be a different library

Reply:
Not if  we keep with_cjson=False by default. As the package ID will be computed with with_cjson=False, in the future, when available, we will have a new option value (different package ID too), but won't require a new option. The introduction of an option generates a new package id.

Reply:
ok, sounds good, will do